matchbox_socket/src/webrtc_socket/socket.rs

```diff
@@ -130,72 +224,21 @@ pub struct WebRtcSocket {
-    /// Create a new connection to the given room with a single unreliable data channel
-    ///
-    /// See [`WebRtcSocketConfig::room_url`] for details on the room url.
-    ///
-    /// The returned future should be awaited in order for messages to be sent and received.
-    #[must_use]
-    pub fn new_unreliable(room_url: impl Into<String>) -> (Self, MessageLoopFuture) {
-        WebRtcSocket::new_with_config(WebRtcSocketConfig {
-            room_url: room_url.into(),
-            ice_server: RtcIceServerConfig::default(),
-            channels: vec![ChannelConfig::unreliable()],
-            attempts: Some(3),
-        })
-    }
-
-    /// Create a new connection to the given room with a single reliable data channel
+    /// Creates a new builder for a connection to a given room with a given number of
+    /// re-connection attempts.
     ///
-    /// See [`WebRtcSocketConfig::room_url`] for details on the room url.
-    ///
-    /// The returned future should be awaited in order for messages to be sent and received.
-    #[must_use]
-    pub fn new_reliable(room_url: impl Into<String>) -> (Self, MessageLoopFuture) {
-        WebRtcSocket::new_with_config(WebRtcSocketConfig {
-            room_url: room_url.into(),
-            ice_server: RtcIceServerConfig::default(),
-            channels: vec![ChannelConfig::reliable()],
-            attempts: Some(3),
-        })
-    }
-
-    /// Create a new connection with the given [`WebRtcSocketConfig`]
-    ///
-    /// The returned future should be awaited in order for messages to be sent and received.
-    #[must_use]
-    pub fn new_with_config(config: WebRtcSocketConfig) -> (Self, MessageLoopFuture) {
-        if config.channels.is_empty() {
-            panic!("You need to configure at least one channel in WebRtcSocketConfig");
-        }
-
-        let (messages_from_peers_tx, messages_from_peers) = new_senders_and_receivers(&config);
-        let (peer_state_tx, peer_state_rx) = futures_channel::mpsc::unbounded();
-        let (peer_messages_out_tx, peer_messages_out_rx) = new_senders_and_receivers(&config);
-
-        let (id_tx, id_rx) = crossbeam_channel::bounded(1);
-
-        (
-            Self {
-                id: Default::default(),
-                id_rx,
-                messages_from_peers,
-                peer_messages_out: peer_messages_out_tx,
-                peer_state_rx,
-                peers: Default::default(),
-            },
-            Box::pin(run_socket(
-                id_tx,
-                config,
-                peer_messages_out_rx,
-                peer_state_tx,
-                messages_from_peers_tx,
-            )),
-        )
+    /// You must add at least one channel with [`WebRtcSocketBuilder::add_channel`],
+    /// [`WebRtcSocketBuilder::add_reliable_channel`], or
+    /// [`WebRtcSocketBuilder::add_unreliable_channel`] before you can build the
+    /// [`WebRtcSocket`]
+    pub fn builder(room_url: impl ToString) -> WebRtcSocketBuilder {
```

Comment:
Why `ToString` instead of `Into<String>`?

Reply:
Nothing specific, should likely be `Into<String>` as the move semantics are more correct